### PR TITLE
Add accommodation and room endpoints

### PR DESCRIPTION
**Pull request by oleg1std**

### CODE REVIEW

File: HappyTravel.Hiroshima.WebApi/Controllers/DirectManager/AccommodationManagementController.cs [hunk 1]
```diff
@@ -83,16 +83,67 @@ public async Task<IActionResult> UpdateAccommodation([FromRoute] string accommod
...
+        
+        /// <summary>
+        /// Removes a accommodation room of by ID
+        /// </summary>
+        /// <param name="accommodationId"></param>
+        /// <param name="roomId"></param>
+        /// <returns></returns>
+        [HttpDelete("{accommodationId}/rooms/{roomId}")]
+        [ProducesResponseType((int) HttpStatusCode.OK)]
+        public async Task<IActionResult> DeleteRoom([FromRoute] int accommodationId, [FromRoute] int roomId)
```
Comment (kirillta) [inline HappyTravel.Hiroshima.WebApi/Controllers/DirectManager/AccommodationManagementController.cs:138]: Use `Remove` if it's not a database-related method

File: HappyTravel.Hiroshima.Common/Models/Accommodations/OccupancyDefinition.cs [hunk 2]
```diff
@@ -2,9 +2,9 @@
 {
     public class OccupancyDefinition
```
Comment (kirillta) [inline HappyTravel.Hiroshima.Common/Models/Accommodations/OccupancyDefinition.cs:3]: The approach with nullable ranges seems strange

File: HappyTravel.Hiroshima.DirectContracts/Services/Availability/RoomAvailabilityService.cs [hunk 3]
```diff
@@ -147,26 +147,30 @@ void AddRoomToDictionaryWithIds(int occupationRequestId, Room room)
             foreach (var childrenAge in occupationRequest.ChildrenAges)
             {
                 if (occupancyDefinition.Infant != null &&
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectContracts/Services/Availability/RoomAvailabilityService.cs:149]: I suggest the `HasValue` condition check

File: HappyTravel.Hiroshima.DirectContracts/Services/Management/AccommodationManagementRepository.cs [hunk 4]
```diff
@@ -26,18 +30,86 @@ public async Task<Accommodation> AddAccommodation(Accommodation accommodation)
...
+        
+        private async Task DeleteAccommodation(int accommodationId)
+        {
+            var accommodationToDelete = new Accommodation {Id = accommodationId};
+            _dbContext.Accommodations.RemoveRange(accommodationToDelete);
+            await _dbContext.SaveChangesAsync();
+        }
+        
+        
+        private void DefineIdAndDetach(List<Room> rooms)
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectContracts/Services/Management/AccommodationManagementRepository.cs:102]: Better to return a list. Void masks the id definition

File: HappyTravel.Hiroshima.Common/Models/Accommodations/OccupancyDefinition.cs (hunk 2 above)
Comment (oleg1std) [reply HappyTravel.Hiroshima.Common/Models/Accommodations/OccupancyDefinition.cs:3]: Some hotels don't have a Teenager Age definition. And some hotels have only an Adult age definition

File: HappyTravel.Hiroshima.DirectManager/RequestValidators/OccupancyConfigurationValidator.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+{
+    public class OccupancyConfigurationValidator : AbstractValidator<OccupancyConfiguration>
+    {
+        public OccupancyConfigurationValidator()
+        {
+            RuleFor(oc => oc.Infants).LessThanOrEqualTo(6);
+            RuleFor(oc => oc.Children).LessThanOrEqualTo(6);
+            RuleFor(oc => oc.Teenagers).LessThanOrEqualTo(6);
+            RuleFor(oc => oc.Adults).LessThanOrEqualTo(6);
+            RuleFor(oc => oc.Infants + oc.Children + oc.Teenagers + oc.Adults).LessThanOrEqualTo(6);
```
[the remark this answers is not in the document]
Comment (oleg1std) [reply HappyTravel.Hiroshima.DirectManager/RequestValidators/OccupancyConfigurationValidator.cs:14]: I don't really know why

[the remark this answers is not in the document]
Comment (oleg1std) [reply HappyTravel.Hiroshima.DirectManager/RequestValidators/OccupancyConfigurationValidator.cs:14]: It's a temporary implementation. I'll add a constant.

File: HappyTravel.Hiroshima.Common/Models/Accommodations/OccupancyDefinition.cs (hunk 2 above)
Comment (kirillta) [reply HappyTravel.Hiroshima.Common/Models/Accommodations/OccupancyDefinition.cs:3]: I know, but the behavior isn't understandable. Me personally prefer approach with a list with a number of definitions, where each of them have its own kind

**Pull request merged**